cli/src/update/unix.rs

```diff
@@ -230,119 +184,57 @@ impl ApplicationUpdater {
 
         // Get the URL for each asset from the Github JSON response in `latest`.
         debug!("Finding the github assets in the Github JSON response");
-        let bin_asset_url = self.find_github_asset(&latest, bin_asset_name)?;
-        let bash_asset_url = self.find_github_asset(&latest, shell_asset_name)?;
-        let minisign_name = format!("{}.minisig", bin_asset_name);
-        let sig_asset_url = self.find_github_asset(&latest, minisign_name.as_str())?;
+        let zip_asset =
+            self.find_github_asset(&latest, format!("{}.zip", archive_name).as_str())?;
+        let sig_asset =
+            self.find_github_asset(&latest, format!("{}.zip.minisig", archive_name).as_str())?;
 
         debug!("Downloading the update files");
-        let bin = self.download_file(bin_asset_url, "phylum.update").await?;
-        let bash = self.download_file(bash_asset_url, shell_asset_name).await?;
-        let sig = self
-            .download_file(sig_asset_url, "phylum.update.minisig")
-            .await?;
-
-        debug!("Verifying the binary signature before move");
-        if !self.has_valid_signature(bin.as_str(), sig.as_str()) {
-            return Err(std::io::Error::new(
-                io::ErrorKind::Other,
-                "The update binary failed signature validation",
-            ));
-        }
+        let zip = download_github_asset(zip_asset).await?;
+        let sig = download_github_asset(sig_asset).await?;
 
-        // If the download and validation succeeds _then_ we move it to overwrite
-        // the existing binary and bash file.
-        debug!("Copying the files to the intended install location");
-        fs::remove_file(&installed_bin_path)?;
-        fs::copy(&bin, &installed_bin_path)?;
-        debug!(
-            "Copying shell script from {} to {:?}",
-            bash, installed_bash_path
-        );
-        fs::copy(&bash, &installed_bash_path)?;
-        debug!("Setting permissions ");
-        fs::set_permissions(&installed_bin_path, fs::Permissions::from_mode(0o770))?;
-        debug!("Removing bin");
-        fs::remove_file(&bin)?;
-        debug!("Removing shell script");
-        fs::remove_file(&bash)?;
-
-        // Ensure that the files copied to the final location were the ones
-        // we expected. This is to address a potential race condition between
-        // the check and the copy.
-        debug!("Verifying the file wasn't changed/tampered with before the move");
-        let final_bin = match installed_bin_path.clone().into_os_string().into_string() {
-            Ok(x) => x,
-            Err(_) => {
-                return Err(std::io::Error::new(
-                    io::ErrorKind::Other,
-                    "Could not create the path for the installation",
-                ))
-            }
-        };
+        debug!("Verifying the package signature");
+        if !self.has_valid_signature(zip.as_ref(), std::str::from_utf8(sig.as_ref())?) {
```

Comment:
```suggestion
        if !self.has_valid_signature(&zip, str::from_utf8(sig.as_ref())?) {
```

I'd suggest importing `str` here, just to keep things concise. Also means in the future nobody has to worry about whether it's worth it to start importing it now, or to also use the fully qualified path so they don't have to touch unrelated code.